bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GroupItem.java

```diff
@@ -310,8 +312,15 @@ public void stateChanged(Item item, State oldState, State newState) {
     @Override
     public void stateUpdated(Item item, State state) {
         State oldState = this.state;
-        if (function != null) {
-            setState(function.calculate(members));
+        if (function != null && baseItem != null) {
+            State calculatedState = function.calculate(members);
+
+            // functions return DecimalTypes so we have to convert them if the baseItem type only supports PercentType

```

Comment:
Good idea, this method is exactly what i was looking for, but I was not aware of that it exists ;) I have changed it now.